### PR TITLE
expose configuration for envoy's RateLimitedAsResourceExhausted

### LINKED ISSUE #4901: Question: Returning custom gRPC header or status in rate limiting
**What question do you have?:**

I'm curious to know if there is a way to set a custom gRPC status code and/or header to be returned if a rate limit is exceeded via the contour rate limiting system, rather than simply returning 429. The 429 error gets mapped by gRPC into an `UNAVAILABLE` error without much user-visible context, and we'd like to provide a nicer error message.

### PR DESCRIPTION
**Pull request by vroldanbet**
Closes https://github.com/projectcontour/contour/issues/4901

The Rate Limit filter in Envoy translates a `429 HTTP` response code to `UNAVAILABLE` as specified in the [gRPC mapping document](https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md). Alternatively, Envoy allows translating it to a less ambiguous `RESOURCE_EXHAUSTED`. 

This commit introduces a new setting to allow contour to forward the same parameter introduced in https://github.com/envoyproxy/envoy/pull/4879

The default value is disabled to retain the original behaviour of returning `UNAVAILABLE`, as changing it would be a breaking change.

This PR is largely inspired in the work done at https://github.com/projectcontour/contour/pull/3457

FYI @sunjayBhatia as we discussed about this in https://github.com/projectcontour/contour/issues/4901

Closes #4901.

### CODE REVIEW

**Comment by skriss** [comment]
Thanks for the PR @vroldanbet! Please run `make generate` and commit the results to get the codegen CI check to pass.

**Comment by skriss** [comment]
Please also address https://github.com/projectcontour/contour/actions/runs/3895937483/jobs/6651861120, the output has instructions to follow. Thanks!

**Comment by vroldanbet** [comment]
@skriss done! Ready for another CI run!

File: examples/contour/01-crds.yaml [hunk 1]
```diff
@@ -1006,7 +1011,7 @@ spec:
                           feature gate. \n This field is immutable."
                         items:
                           x-kubernetes-map-type: atomic
-                          description: ResourceClaim references one entry in PodSpec.ResourceClaims.
+description: ResourceClaim references one entry in PodSpec.ResourceClaims.
```
Comment (skriss) [inline examples/contour/01-crds.yaml:1014]: We just merged https://github.com/projectcontour/contour/pull/4972 to fix this whitespace problem for macOS, if you could rebase your branch on `main` and re-run `make generate` it should go away.

**Review by skriss — CHANGES_REQUESTED** [review]

**Comment by skriss** [comment]
Looks like one UT still needs to be updated for the new field (https://github.com/projectcontour/contour/blob/main/cmd/contour/servecontext_test.go#L623-L645), otherwise this looks good to me

**Comment by vroldanbet** [comment]
@skriss just pushed the fix, thanks for your patience and prompt feedback 🙏🏻

**Review by sunjayBhatia — APPROVED** [review]

**Review by skriss — APPROVED** [review]
LGTM, thanks @vroldanbet!

**Pull request merged**